Move focus ring code to OutlinePainter

anttijk opened the pull request:
#### 8562be651f6b2ce9d5cf4ed9b8bed1c45c821c37
<pre>
Move focus ring code to OutlinePainter
<a href="https://bugs.webkit.org/show_bug.cgi?id=303168">https://bugs.webkit.org/show_bug.cgi?id=303168</a>
<a href="https://rdar.apple.com/165474109">rdar://165474109</a>

Reviewed by Alan Baradlay.

Move more code out of RenderObjects.

* Source/WebCore/rendering/BorderPainter.cpp:
* Source/WebCore/rendering/OutlinePainter.cpp:
(WebCore::OutlinePainter::paintOutline const):
(WebCore::usePlatformFocusRingColorForOutlineStyleAuto):
(WebCore::useShrinkWrappedFocusRingForOutlineStyleAuto):
(WebCore::drawFocusRing):
(WebCore::OutlinePainter::paintFocusRing const):
(WebCore::OutlinePainter::collectFocusRingRects):
(WebCore::OutlinePainter::collectFocusRingRectsForListBox):
(WebCore::OutlinePainter::collectFocusRingRectsForInline):
(WebCore::OutlinePainter::collectFocusRingRectsForBlock):
(WebCore::OutlinePainter::collectFocusRingRectsForChildBox):
(WebCore::OutlinePainter::collectFocusRingRectsForInlineChildren):
(WebCore::OutlinePainter::deviceScaleFactor const):
* Source/WebCore/rendering/OutlinePainter.h:
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::addFocusRingRectsForInlineChildren const): Deleted.
(WebCore::RenderBlock::addFocusRingRects const): Deleted.
(WebCore::RenderBlock::addFocusRingRectsForBlockChild const): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::addFocusRingRectsForInlineChildren const): Deleted.
* Source/WebCore/rendering/RenderBlockFlow.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::issueRepaintForOutlineAuto):
(WebCore::usePlatformFocusRingColorForOutlineStyleAuto): Deleted.
(WebCore::useShrinkWrappedFocusRingForOutlineStyleAuto): Deleted.
(WebCore::drawFocusRing): Deleted.
(WebCore::RenderElement::paintFocusRing const): Deleted.
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::paintRectToClipOutFromBorder):
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::collectLineBoxRects const):
(WebCore::RenderInline::paintOutline const):
(WebCore::RenderInline::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderListBox.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::addPDFURLRect const):
(WebCore::RenderObject::absoluteFocusRingQuads):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderTextControl.h:
* Source/WebCore/rendering/svg/RenderSVGModelObject.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.cpp:
(WebCore::LegacyRenderSVGModelObject::addFocusRingRects const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.h:

Canonical link: <a href="https://commits.webkit.org/303606@main">https://commits.webkit.org/303606@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fe81c1ccf53e51fdbf04e873973469510b00e697

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132977 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5478 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44072 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140512 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d7b4f0f0-70c9-4d63-af91-26af48d5432c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134847 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5904 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5341 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/101684 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/72b5f0b0-0dc7-4877-a358-4b0c4dd9ac84) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135923 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/4141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/119153 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82483 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/560a8bfa-f925-46d2-8821-ffd86d2b5e5e) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/4036 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1647 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83747 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/113076 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37272 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143167 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5147 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/37852 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/110068 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5229 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/4407 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110248 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27948 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3947 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115415 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/58718 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5201 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/33769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5042 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68653 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5291 "Built successfully") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/5159 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderObject.cpp:
```diff
@@ -877,13 +882,17 @@ IntRect RenderObject::absoluteBoundingBoxRect(bool useTransforms, bool* wasFixed
     Vector<LayoutRect> rects;
     // FIXME: addFocusRingRects() needs to be passed this transform-unaware
-    // localToAbsolute() offset here because RenderInline::addFocusRingRects()
+    // localToAbsolute() offset here because BorderPainter::addFocusRingRects()
```
BorderPainter::addFocusRingRects()?

anttijk committed:
Move focus ring code to OutlinePainter
https://bugs.webkit.org/show_bug.cgi?id=303168
rdar://165474109

Reviewed by Alan Baradlay.

Move more code out of RenderObjects.

* Source/WebCore/rendering/BorderPainter.cpp:
* Source/WebCore/rendering/OutlinePainter.cpp:
(WebCore::OutlinePainter::paintOutline const):
(WebCore::usePlatformFocusRingColorForOutlineStyleAuto):
(WebCore::useShrinkWrappedFocusRingForOutlineStyleAuto):
(WebCore::drawFocusRing):
(WebCore::OutlinePainter::paintFocusRing const):
(WebCore::OutlinePainter::collectFocusRingRects):
(WebCore::OutlinePainter::collectFocusRingRectsForListBox):
(WebCore::OutlinePainter::collectFocusRingRectsForInline):
(WebCore::OutlinePainter::collectFocusRingRectsForBlock):
(WebCore::OutlinePainter::collectFocusRingRectsForChildBox):
(WebCore::OutlinePainter::collectFocusRingRectsForInlineChildren):
(WebCore::OutlinePainter::deviceScaleFactor const):
* Source/WebCore/rendering/OutlinePainter.h:
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::addFocusRingRectsForInlineChildren const): Deleted.
(WebCore::RenderBlock::addFocusRingRects const): Deleted.
(WebCore::RenderBlock::addFocusRingRectsForBlockChild const): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::addFocusRingRectsForInlineChildren const): Deleted.
* Source/WebCore/rendering/RenderBlockFlow.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::issueRepaintForOutlineAuto):
(WebCore::usePlatformFocusRingColorForOutlineStyleAuto): Deleted.
(WebCore::useShrinkWrappedFocusRingForOutlineStyleAuto): Deleted.
(WebCore::drawFocusRing): Deleted.
(WebCore::RenderElement::paintFocusRing const): Deleted.
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::paintRectToClipOutFromBorder):
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::collectLineBoxRects const):
(WebCore::RenderInline::paintOutline const):
(WebCore::RenderInline::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderListBox.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::addPDFURLRect const):
(WebCore::RenderObject::absoluteFocusRingQuads):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::addFocusRingRects const): Deleted.
* Source/WebCore/rendering/RenderTextControl.h:
* Source/WebCore/rendering/svg/RenderSVGModelObject.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.cpp:
(WebCore::LegacyRenderSVGModelObject::addFocusRingRects const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.h:

Canonical link: https://commits.webkit.org/303606@main
